inception/inception-api/src/main/java/de/tudarmstadt/ukp/clarin/webanno/api/ProjectService.java

```diff
@@ -484,4 +476,56 @@ static MDCContext withProjectLogger(Project aProject)
 
         return MDCContext.open().with(KEY_PROJECT_ID, String.valueOf(aProject.getId()));
     }
+
+    String deriveSlugFromName(String aName);
+
+    String deriveUniqueSlug(String aSlug);
+
+    /**
+     * Check if the name is valid, SPecial characters are not allowed as a project/user name as it
+     * will conflict with file naming system
+     * 
+     * @param aName
+     *            a name.
+     * @return if the name is valid.
+     */
+    static boolean isValidProjectName(String aName)
+    {
+        return aName != null && containsNone(aName, PROJECT_NAME_ILLEGAL_CHARACTERS);
+    }
+
+    static boolean isValidProjectSlug(String aSlug)
+    {
+        if (aSlug == null || aSlug.isEmpty()) {
+            return false;
+        }
+
+        if (aSlug.length() < MIN_PROJECT_SLUG_LENGTH || aSlug.length() > MAX_PROJECT_SLUG_LENGTH) {
+            return false;
+        }
+
+        // Must start with a letter character
+        if (!isValidProjectSlugInitialCharacter(aSlug.charAt(0))) {
+            return false;
+        }
+
+        // Must consist only of valid characters
+        for (int i = 0; i < aSlug.length(); i++) {
+            if (!isValidProjectSlugCharacter(aSlug.charAt(i))) {
```

Comment:
StringUtils.containsNone?

Reply:
Nah, containsNone does not work - I can't sensibly enumerate all but `[a-Z][0-9]-_`. I'd have to use "containsOnly" but then still would have to enumerate all the characters. Writing a custom predicate is better.